js/ui/marker.js

```diff
@@ -13,6 +13,10 @@ var Point = require('point-geometry');
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
  * @param {Object=} options
  * @param {PointLike=} options.offset The offset in pixels as a [`PointLike`](#PointLike) object to apply relative to the element's top left corner. Negatives indicate left and up.
+ * @param {boolean=} [options.pitchScale=false] Whether the marker should scale smaller into the background or larger into the foreground when the map is pitched and rotated
+ * @param {boolean=} [options.zoomScale=false] Whether the marker should scale with the map as you zoom in and out
+ * @param {number=} [options.nativeZoom=map.getZoom()] The native zoom level of the marker. If the map is at this zoom the marker will not be scaled. If zoom is greater the marker will be enlarged. If zoom is less the marker will be shrunk.
+ * @param {number=} [options.zoomFactor=1.6] The multiple by which to scale the size of the marker between each zoom level
```

Comment:
Strictly speaking, in order to get the marker to scale with the ma 1:1 in the zoom range `[1, 20]` all you'd need to do is
```
stops: [
    [1, 1/1024]
    [20, 1024]
]
```

I agree the existing function syntax specification does not lend itself to an elegant implementation of your use case. That's a problem with the existing specification, not a reason to not use the existing specification. 

I'd be happy to entertain proposals for extending the function syntax or helper functions to generate function stops with certain characteristics.